### DISCUSSION TITLE
Error generating database

### DISCUSSION THREAD

**Discussion by id77777777** [shlinkio/shlink#1864] [Help wanted] [answered]
### Shlink version

3.10.2

### PHP version

Included

### How do you serve Shlink

RoadRunner Docker image

### Database engine

MariaDB

### Database version

10.8

### Summary

Hi All,

I'm getting "Oops! Could not connect to this Shlink server."

Shlink GUI Logs
`     
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf differs from the packaged version
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-shlink-servers-json.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2023/08/30 16:02:31 [notice] 1#1: using the "epoll" event method
2023/08/30 16:02:31 [notice] 1#1: nginx/1.23.4
2023/08/30 16:02:31 [notice] 1#1: built by gcc 12.2.1 20220924 (Alpine 12.2.1_git20220924-r4) 
2023/08/30 16:02:31 [notice] 1#1: OS: Linux 5.4.0-139-generic
2023/08/30 16:02:31 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2023/08/30 16:02:31 [notice] 1#1: start worker processes
2023/08/30 16:02:31 [notice] 1#1: start worker process 30
2023/08/30 16:02:31 [notice] 1#1: start worker process 31
000.000.000.000 - - [30/Aug/2023:16:02:35 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:02:41 +0000] "GET /server/812995ed-277f-445e-a0a6-APIHIDDEN HTTP/1.1" 200 6093 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:02:42 +0000] "GET /favicon.svg HTTP/1.1" 200 2248 "http://XX.XX.XX.XXX:9090/server/812995ed-277f-445e-a0a6-APIHIDDEN" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:03:42 +0000] "GET /server/812995ed-277f-445e-a0a6-APIHIDDEN HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:03:50 +0000] "OPTIONS /rest/v3/health HTTP/1.1" 405 559 "http://XX.XX.XX.XXX:9090/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:22 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:22 +0000] "GET /assets/index-14f07603.css HTTP/1.1" 200 254953 "http://domain.online(hidden):9090/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:22 +0000] "GET /assets/index-5c19fb17.js HTTP/1.1" 200 1462457 "http://domain.online(hidden):9090/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
2023/08/30 16:06:22 [error] 30#30: *5 open() "/usr/share/nginx/html/conf.d/servers.json" failed (2: No such file or directory), client: 000.000.000.000, server: , request: "GET /servers.json HTTP/1.1", host: "domain.online(hidden):9090", referrer: "http://domain.online(hidden):9090/"
000.000.000.000 - - [30/Aug/2023:16:06:22 +0000] "GET /servers.json HTTP/1.1" 404 555 "http://domain.online(hidden):9090/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:52 +0000] "GET /server/812995ed-277f-445e-a0a6-APIHIDDEN HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:55 +0000] "GET /server/5e2b18f8-4e42-4cbd-a355-225395d4ee98 HTTP/1.1" 200 6093 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:06:55 +0000] "GET /favicon.svg HTTP/1.1" 200 2248 "http://domain.online(hidden):9090/server/5e2b18f8-4e42-4cbd-a355-225395d4ee98" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"
000.000.000.000 - - [30/Aug/2023:16:13:54 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.54" "-"`

Shlink Backend Logs

`Initializing database if needed...
 [ERROR] Error generating database.. Run this command with -vvv to see specific 
         error info.                                                            
Initializing database if needed... Success!
Updating database... Success!
Generating proxies... Success!
Clearing entities cache... Success!
Downloading GeoLite2 db file... Success!
DBG ts=1693411356158069200 logger=rpc msg=plugin was started address=tcp://127.0.0.1:6001 list of the plugins with RPC methods:=["lock","app","informer","resetter","jobs"]
DBG ts=1693411356158578000 logger=jobs msg=driver ready pipeline=shlink driver=memory start=1693411356158347800 elapsed=0.000228771
DBG ts=1693411356158635500 logger=memory msg=pipeline was started driver=memory pipeline=shlink start=2023-08-30 16:02:36.158630528 +0000 UTC elapsed=8.367µs
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 34, "internal_event_name": "EventWorkerConstruct"}
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 35, "internal_event_name": "EventWorkerConstruct"}
[INFO] RoadRunner server started; version: 2023.1.5, buildtime: 2023-06-08T14:45:05+0000
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 40, "internal_event_name": "EventWorkerConstruct"}
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 41, "internal_event_name": "EventWorkerConstruct"}
DBG ts=1693411579167607800 logger=jobs msg=job was pushed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 pipeline=shlink driver=memory start=1693411579167411700 elapsed=0.000185485
DBG ts=1693411579167743200 logger=memory msg=message pushed to the priority queue current=1 limit=10
DBG ts=1693411579167819500 logger=jobs msg=job processing was started ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.684336263
DBG ts=1693411579170880300 logger=jobs msg=job was pushed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc pipeline=shlink driver=memory start=1693411579170813200 elapsed=0.00006476
DBG ts=1693411579171301600 logger=memory msg=message pushed to the priority queue current=2 limit=10
DBG ts=1693411579171507500 logger=jobs msg=job processing was started ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.68787252
DBG ts=1693411579179282200 logger=jobs msg=job was pushed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 pipeline=shlink driver=memory start=1693411579179206700 elapsed=0.000073425
DBG ts=1693411579179570200 logger=memory msg=message pushed to the priority queue current=3 limit=10
DBG ts=1693411579179617800 logger=jobs msg=job processing was started ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.695979883
DBG ts=1693411579180382200 logger=jobs msg=job was pushed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 pipeline=shlink driver=memory start=1693411579180291600 elapsed=0.000088713
DBG ts=1693411579180566500 logger=memory msg=message pushed to the priority queue current=4 limit=10
DBG ts=1693411579180643800 logger=jobs msg=job processing was started ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.696998643
DBG ts=1693411579181392400 logger=jobs msg=job was pushed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c pipeline=shlink driver=memory start=1693411579181353500 elapsed=0.000037826
DBG ts=1693411579181662700 logger=memory msg=message pushed to the priority queue current=5 limit=10
2023-08-30T16:06:19+0000	INFO	server      	[2023-08-30T16:06:19.230188+00:00] [7d251dab-dd7d-4362-8331-5be27d69b2a5] Access.INFO - GET / 404 979
DBG ts=1693411579254555600 logger=jobs msg=job was processed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.771070293
DBG ts=1693411579255134200 logger=jobs msg=job processing was started ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.000094451
DBG ts=1693411579262139400 logger=jobs msg=job was processed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.778500322
DBG ts=1693411579276876500 logger=jobs msg=job was processed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.793239684
DBG ts=1693411579285979400 logger=jobs msg=job was processed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.802331917
DBG ts=1693411579302071600 logger=jobs msg=job was processed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.047029139`

MariaDB Logs

`Initializing database if needed...
 [ERROR] Error generating database.. Run this command with -vvv to see specific 
         error info.                                                            
Initializing database if needed... Success!
Updating database... Success!
Generating proxies... Success!
Clearing entities cache... Success!
Downloading GeoLite2 db file... Success!
DBG ts=1693411356158069200 logger=rpc msg=plugin was started address=tcp://127.0.0.1:6001 list of the plugins with RPC methods:=["lock","app","informer","resetter","jobs"]
DBG ts=1693411356158578000 logger=jobs msg=driver ready pipeline=shlink driver=memory start=1693411356158347800 elapsed=0.000228771
DBG ts=1693411356158635500 logger=memory msg=pipeline was started driver=memory pipeline=shlink start=2023-08-30 16:02:36.158630528 +0000 UTC elapsed=8.367µs
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 34, "internal_event_name": "EventWorkerConstruct"}
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 35, "internal_event_name": "EventWorkerConstruct"}
[INFO] RoadRunner server started; version: 2023.1.5, buildtime: 2023-06-08T14:45:05+0000
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 40, "internal_event_name": "EventWorkerConstruct"}
2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 41, "internal_event_name": "EventWorkerConstruct"}
DBG ts=1693411579167607800 logger=jobs msg=job was pushed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 pipeline=shlink driver=memory start=1693411579167411700 elapsed=0.000185485
DBG ts=1693411579167743200 logger=memory msg=message pushed to the priority queue current=1 limit=10
DBG ts=1693411579167819500 logger=jobs msg=job processing was started ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.684336263
DBG ts=1693411579170880300 logger=jobs msg=job was pushed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc pipeline=shlink driver=memory start=1693411579170813200 elapsed=0.00006476
DBG ts=1693411579171301600 logger=memory msg=message pushed to the priority queue current=2 limit=10
DBG ts=1693411579171507500 logger=jobs msg=job processing was started ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.68787252
DBG ts=1693411579179282200 logger=jobs msg=job was pushed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 pipeline=shlink driver=memory start=1693411579179206700 elapsed=0.000073425
DBG ts=1693411579179570200 logger=memory msg=message pushed to the priority queue current=3 limit=10
DBG ts=1693411579179617800 logger=jobs msg=job processing was started ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.695979883
DBG ts=1693411579180382200 logger=jobs msg=job was pushed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 pipeline=shlink driver=memory start=1693411579180291600 elapsed=0.000088713
DBG ts=1693411579180566500 logger=memory msg=message pushed to the priority queue current=4 limit=10
DBG ts=1693411579180643800 logger=jobs msg=job processing was started ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.696998643
DBG ts=1693411579181392400 logger=jobs msg=job was pushed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c pipeline=shlink driver=memory start=1693411579181353500 elapsed=0.000037826
DBG ts=1693411579181662700 logger=memory msg=message pushed to the priority queue current=5 limit=10
2023-08-30T16:06:19+0000	INFO	server      	[2023-08-30T16:06:19.230188+00:00] [7d251dab-dd7d-4362-8331-5be27d69b2a5] Access.INFO - GET / 404 979
DBG ts=1693411579254555600 logger=jobs msg=job was processed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.771070293
DBG ts=1693411579255134200 logger=jobs msg=job processing was started ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.000094451
DBG ts=1693411579262139400 logger=jobs msg=job was processed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.778500322
DBG ts=1693411579276876500 logger=jobs msg=job was processed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.793239684
DBG ts=1693411579285979400 logger=jobs msg=job was processed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.802331917
DBG ts=1693411579302071600 logger=jobs msg=job was processed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.047029139`

All logs seem to be fine as far as I am aware. I've ran all three in Portainer containers using a Stack. Apologies formatting seems to be awful!

![image](https://github.com/shlinkio/shlink/assets/132589476/c81aafa0-92dc-42dd-b5c3-617458024ecf)

Please could someone point me in the right direction?

**Comment by acelaya** [comment]
Seems to be an error with the database connection. Try providing `SHELL_VERBOSITY=3` https://shlink.io/documentation/troubleshooting/#troubleshoot-errors-when-booting-up-the-docker-image

**Reply by id77777777** [reply]
Thanks for getting back to me so quickly.

`Initializing database if needed... [ERROR] Error generating database.. Run this command with -vvv to see specific  error info.                                                             Initializing database if needed... Success! Updating database... Success! Generating proxies... Success! Clearing entities cache... Success! Downloading GeoLite2 db file... Success! DBG ts=1693411356158069200 logger=rpc msg=plugin was started address=tcp://127.0.0.1:6001 list of the plugins with RPC methods:=["lock","app","informer","resetter","jobs"] DBG ts=1693411356158578000 logger=jobs msg=driver ready pipeline=shlink driver=memory start=1693411356158347800 elapsed=0.000228771 DBG ts=1693411356158635500 logger=memory msg=pipeline was started driver=memory pipeline=shlink start=2023-08-30 16:02:36.158630528 +0000 UTC elapsed=8.367µs 2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 34, "internal_event_name": "EventWorkerConstruct"} 2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 35, "internal_event_name": "EventWorkerConstruct"} [INFO] RoadRunner server started; version: 2023.1.5, buildtime: 2023-06-08T14:45:05+0000 2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 40, "internal_event_name": "EventWorkerConstruct"} 2023-08-30T16:02:36+0000	DEBUG	server      	worker is allocated	{"pid": 41, "internal_event_name": "EventWorkerConstruct"} DBG ts=1693411579167607800 logger=jobs msg=job was pushed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 pipeline=shlink driver=memory start=1693411579167411700 elapsed=0.000185485 DBG ts=1693411579167743200 logger=memory msg=message pushed to the priority queue current=1 limit=10 DBG ts=1693411579167819500 logger=jobs msg=job processing was started ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.684336263 DBG ts=1693411579170880300 logger=jobs msg=job was pushed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc pipeline=shlink driver=memory start=1693411579170813200 elapsed=0.00006476 DBG ts=1693411579171301600 logger=memory msg=message pushed to the priority queue current=2 limit=10 DBG ts=1693411579171507500 logger=jobs msg=job processing was started ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.68787252 DBG ts=1693411579179282200 logger=jobs msg=job was pushed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 pipeline=shlink driver=memory start=1693411579179206700 elapsed=0.000073425 DBG ts=1693411579179570200 logger=memory msg=message pushed to the priority queue current=3 limit=10 DBG ts=1693411579179617800 logger=jobs msg=job processing was started ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.695979883 DBG ts=1693411579180382200 logger=jobs msg=job was pushed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 pipeline=shlink driver=memory start=1693411579180291600 elapsed=0.000088713 DBG ts=1693411579180566500 logger=memory msg=message pushed to the priority queue current=4 limit=10 DBG ts=1693411579180643800 logger=jobs msg=job processing was started ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.696998643 DBG ts=1693411579181392400 logger=jobs msg=job was pushed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c pipeline=shlink driver=memory start=1693411579181353500 elapsed=0.000037826 DBG ts=1693411579181662700 logger=memory msg=message pushed to the priority queue current=5 limit=10 2023-08-30T16:06:19+0000	INFO	server      	[2023-08-30T16:06:19.230188+00:00] [7d251dab-dd7d-4362-8331-5be27d69b2a5] Access.INFO - GET / 404 979 DBG ts=1693411579254555600 logger=jobs msg=job was processed successfully ID=618c1437-8729-437c-aaa5-1c9cddc2f131 start=1693411356483481600 elapsed=222.771070293 DBG ts=1693411579255134200 logger=jobs msg=job processing was started ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.000094451 DBG ts=1693411579262139400 logger=jobs msg=job was processed successfully ID=1c6c88ed-2c68-4f12-bed5-9f37995789a5 start=1693411356483636700 elapsed=222.778500322 DBG ts=1693411579276876500 logger=jobs msg=job was processed successfully ID=f93d8bd5-6f19-4526-bb77-f3b430ecefbc start=1693411356483633700 elapsed=222.793239684 DBG ts=1693411579285979400 logger=jobs msg=job was processed successfully ID=188f1dfe-a289-4920-ab57-cea788f126a3 start=1693411356483644200 elapsed=222.802331917 DBG ts=1693411579302071600 logger=jobs msg=job was processed successfully ID=f7a7aabc-0f29-4137-b297-47e711a0cd3c start=1693411579255038500 elapsed=0.047029139`

Here is the updated logs for Shlink backend with VERBOSITY on 3. I have a feeling this is related to me deploying the stack over and over again, maybe something already exists somewhere...

**Reply by acelaya** [reply]
Those logs are still the same. Make sure you set `SHELL_VERBOSITY`, not `VERBOSITY`.

**Reply by id77777777** [reply]
Hi,

![image](https://github.com/shlinkio/shlink/assets/132589476/072f9b11-0d72-42ea-91b6-8340d671c1fe)

I think I may have fixed by re-deploying so apologies for that.

Logs: (https://pastebin.com/L6uJrL2h)

The database exists, but it doesn't exist... I know this isn't a problem with Shlink so apologies but do you know how to remove a stale docker database from the "void"

**Reply by id77777777** [reply]
I'm also getting the following constantly.

`2023-09-05 13:49:13 158 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:50:14 159 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:50:14 160 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:50:14 161 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:51:15 162 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:51:15 163 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:51:15 164 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:52:16 165 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:52:16 166 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:52:16 167 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:53:17 168 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:53:17 169 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:53:17 170 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:54:18 171 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:54:18 172 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:54:18 173 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:55:19 174 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:55:19 175 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:55:19 176 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:56:20 177 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:56:20 178 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)
2023-09-05 13:56:20 179 [Warning] Access denied for user 'nxitadm'@'172.25.0.3' (using password: YES)`

**Reply by acelaya** [reply]
That means you are not providing the right database credentials to Shlink

**Reply by id77777777** [reply]
Well, the thing is, I am... It's very strange. No extra spaces, nothing, it's perfect, however the root password is different, I guess this is fine.

**Reply by id77777777** [reply]
Okay I managed to fix this issue by binding the IP. Also I had to re-assign to different port for some reason on Shlink docker, this seems to be a issue with Docker having ghosted sessions.

However after all is setup, I have an issue. URLs aren't being re-directed. When clicking on a link that was shortened, it goes no where.

**Reply by id77777777** [reply]
QR Codes also don't seem to be working.
![image](https://github.com/shlinkio/shlink/assets/132589476/3264be23-a455-42a2-89bd-f3ce30f77f4a)

**Reply by id77777777** [reply, marked as answer]
I'm going to read about Reverse Proxy, thanks for the speedy replies though!
